charts/csi-secret-provider-class/templates/deployment.yaml

```diff
@@ -0,0 +1,46 @@
+{{- if .Values.dummyDeployment.enabled }}
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
Could this be one of the rare edge cases where we consider directly managing a Pod rather than using a Deployment?

Reply:
We just have to think about the Deploymentstrategy. What does need to happen on changes? Remove the Pod and recreate it? The secret gets removed for a brief short of time. This is why I thought about using the Deployment, but haven't thought about Rollingupdate and so on yet.

Reply:
You are right, the changes will be much more straight forward if it's a Deployment.